web command - collect all roles for an AWS Fed App (idp) at once

monde opened the pull request:
### Collect all roles for an AWS Fed App (IdP) at once

Based on @daniel-sampliner`s work in #94

`okta-aws-cli` will collect all available AWS IAM Roles for a given Okta AWS
Federation app (IdP) at once.  This is a feature specific to writing the
`$HOME/.aws/credentials` file. Roles will be AWS account alias name (if STS list
aliases is available on the given role) then `-` then abbreviated role name.

```
# AWS account alias "myorg", given IdP associated with "AWS Account Federation"
# and an app associated with two roles.

$ okta-aws-cli web \
    --org-domain test.okta.com \
    --oidc-client-id 0oa5wyqjk6Wm148fE1d7 \
    --write-aws-credentials \
    --all-profiles

? Choose an IdP: AWS Account Federation
Updated profile "myorg-S3-read" in credentials file "/Users/me/.aws/credentials".
Updated profile "myorg-S3-write" in credentials file "/Users/me/.aws/credentials".
```

duytiennguyen-okta reviewed (changes requested):
Everything looks fine. I just have concerned with the way you use the Credential interface

duytiennguyen-okta commented on internal/output/aws_credentials_file.go:
```diff
@@ -179,15 +183,16 @@ func NewAWSCredentialsFile(legacyVars bool, expiryVars bool, expiry string) *AWS
 // credentials file.
-func (e *AWSCredentialsFile) Output(c *config.Config, oc *oaws.Credential, ac *sts.Credentials) error {
+func (a *AWSCredentialsFile) Output(c *config.Config, oc oaws.Credential) error {
+	cfc := oc.(*oaws.CredsFileCredential)
```
I really don't like that we have to cast things into concrete type from an interface{}. That has already introduce a lot of issue in the tf provider. Even in the v3 of tfsdk framework, they removed that. I would much rather you defined whatever field you need in the interface and exposing them that way

monde commented:
This was released in v2.0.0-beta.3 but the feature is not complete. The feature is being refined in v2.0.0-beta.4

monde commented:
```
go run cmd/okta-aws-cli/main.go web --oidc-client-id abc --org-domain test.oktapreview.com --open-browser --all-profiles --write-aws-credentials

https://test.oktapreview.com/activate?user_code=QLBKHCCB

Updated profile "devorg-idp1-role1" in credentials file "/Users/me/.aws/credentials".
Updated profile "devorg-idp1-role2" in credentials file "/Users/me/.aws/credentials".
Updated profile "devorg-idp2-role1" in credentials file "/Users/me/.aws/credentials".
Updated profile "prodorg-idp1-role1" in credentials file "/Users/me/.aws/credentials".
```
